__tests__/listener.test.js

```diff
@@ -0,0 +1,81 @@
+const Listener = require('../src/server/Listener');
+
+describe('listener', () => {
+	let listener;
+	beforeEach(() => {
+		listener = new Listener(3003);
+	});
+
+	afterEach(() => {
+		listener.destroy();
+	});
+
+	it('should match paths without tokens', () => {
+		const tokenlessPath = '/path/without/tokens';
+		listener.add({
+			method: 'GET',
+			response: {},
+			uri: tokenlessPath
+		});
+		const result = listener.get(tokenlessPath, 'GET');
+		expect(result.options.uri).toBe(tokenlessPath);
+	});
+
+	it('should match paths with a single token', () => {
+		const tokenPath = '/path/:with/tokens';
+		listener.add({
+			method: 'GET',
+			response: {},
+			uri: tokenPath
+		});
+		const result = listener.get('/path/alpha/tokens', 'GET');
+		expect(result.options.uri).toBe(tokenPath);
+	});
+
+	it('should match paths with multiple tokens', () => {
+		const tokenPath = '/path/:with/tokens/:and/so/:much/other/:stuff';
+		listener.add({
+			method: 'GET',
+			response: {},
+			uri: tokenPath
+		});
+		const result = listener.get('/path/first/tokens/second/so/third/other/fourth', 'GET');
+		expect(result.options.uri).toBe(tokenPath);
+	});
+
+	it('should do partial token matching', () => {
+		const tokenPath = '/path/:with/tokens/:and/:stuff';
+		listener.add({
+			method: 'GET',
+			response: {},
+			uri: tokenPath
+		});
+		const result = listener.get('/path/first/tokens/second', 'GET');
+		expect(result).toBeUndefined();
+	});
```

Comment:
echoing what @gogozby said:

> To test `should do partial matching`, do we need to have this kind of tokenPath: `'/path/:with/tokens/:and/:stuff?'`?

It seems like there is a mismatch between the description, and the expectation.